bin/zkServer.sh

```diff
@@ -236,43 +223,41 @@ restart)
 status)
     # -q is necessary on some versions of linux where nc returns too quickly, and no stat result is output
     isSSL="false"
-    clientPortAddress=`$GREP "^[[:space:]]*clientPortAddress[^[:alpha:]]" "$ZOOCFG" | sed -e 's/.*=//'`
-    if ! [ $clientPortAddress ]
-    then
-	      clientPortAddress="localhost"
+    clientPortAddress=$($GREP "^[[:space:]]*clientPortAddress[^[:alpha:]]" "$ZOOCFG" | sed -e 's/.*=//')
+    if ! [ "$clientPortAddress" ]; then
+      clientPortAddress="localhost"
     fi
-    clientPort=`$GREP "^[[:space:]]*clientPort[^[:alpha:]]" "$ZOOCFG" | sed -e 's/.*=//'`
-    if ! [[ "$clientPort"  =~ ^[0-9]+$ ]]
-    then
-      dataDir=`$GREP "^[[:space:]]*dataDir" "$ZOOCFG" | sed -e 's/.*=//'`
-      myid=`cat "$dataDir/myid" 2> /dev/null`
+    clientPort=$($GREP "^[[:space:]]*clientPort[^[:alpha:]]" "$ZOOCFG" | sed -e 's/.*=//')
+    if ! [[ "$clientPort"  =~ ^[0-9]+$ ]]; then
```

Comment:
This is the first time I've seen the use of the double-square brackets in this code review for a test block. It probably doesn't matter, but FWIW, if you're writing bash, it's best to use the double-square everywhere rather than the single-square braces, because then there's no ambiguity between the bash language syntax and the literal command `/usr/bin/[`. Many people don't realize that `/usr/bin/[` is actually a separate process, although I believe bash just uses its own built-in anyway, so it doesn't have the loss of performance for executing a separate process, but I'm not sure about that. In general, I prefer double over single, if you are using bash. However, I'm not sure it's worth changing this throughout the script.

Reply:
I prefer double square brackets vs single one, but here the only change is replacing the old `Command` by $(Command). The command itself has not been changed at all

Reply:
Right. My comment was inspired by the change I noticed here, but does not pertain to this change. Rather, it pertains to all the other occurrences in the script where single square brackets are used. It's up to you whether you want to migrate everything to double-square brackets in this PR or a subsequent or not at all. :smiley_cat: